plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -825,29 +829,67 @@ protected boolean isSupportedJoinCondition(JdbcJoinCondition joinCondition)
         return true;
     }
 
-    private static ColumnMapping charColumnMapping(int charLength)
+    private static ColumnMapping charColumnMapping(int charLength, boolean withCollate)
     {
         if (charLength > CharType.MAX_LENGTH) {
-            return varcharColumnMapping(charLength);
+            return varcharColumnMapping(charLength, withCollate);
         }
         CharType charType = createCharType(charLength);
         return ColumnMapping.sliceMapping(
                 charType,
                 charReadFunction(charType),
-                charWriteFunction(),
-                POSTGRESQL_CHARACTER_PUSHDOWN);
+                withCollate ? charWriteFunctionWithCollate() : charWriteFunction(),
+                withCollate ? FULL_PUSHDOWN : POSTGRESQL_CHARACTER_PUSHDOWN);
+    }
+
+    private static SliceWriteFunction charWriteFunctionWithCollate()
+    {
+        return new SliceWriteFunction()
+        {
+            @Override
+            public String getBindExpression()
+            {
+                return "? COLLATE \"C\"";
+            }
+
+            @Override
+            public void set(PreparedStatement statement, int index, Slice value)
+                    throws SQLException
+            {
+                statement.setString(index, value.toStringUtf8());
+            }
+        };
     }
 
-    private static ColumnMapping varcharColumnMapping(int varcharLength)
+    private static ColumnMapping varcharColumnMapping(int varcharLength, boolean withCollate)
     {
         VarcharType varcharType = varcharLength <= VarcharType.MAX_LENGTH
                 ? createVarcharType(varcharLength)
                 : createUnboundedVarcharType();
         return ColumnMapping.sliceMapping(
                 varcharType,
                 varcharReadFunction(varcharType),
-                varcharWriteFunction(),
-                POSTGRESQL_CHARACTER_PUSHDOWN);
+                withCollate ? varcharWriteFunctionWithCollate() : varcharWriteFunction(),
+                withCollate ? FULL_PUSHDOWN : POSTGRESQL_CHARACTER_PUSHDOWN);
```

Comment:
POSTGRESQL_CHARACTER_PUSHDOWN needs rename, since it's no longer what the name suggests it is

Reply:
How about `POSTGRESQL_STRING_PUSHDOWN_WITHOUT_COLLATE`?